assertj-core/src/test/java/org/assertj/core/api/iterator/IteratorAssert_isUnmodifiable_Test.java

```diff
@@ -0,0 +1,125 @@
+import org.assertj.core.test.jdk11.Jdk11;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.*;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatNoException;
+import static org.assertj.core.api.BDDAssertions.then;
+import static org.assertj.core.error.ShouldBeUnmodifiable.shouldBeUnmodifiable;
+import static org.assertj.core.error.ShouldNotBeNull.shouldNotBeNull;
+import static org.assertj.core.util.AssertionsUtil.expectAssertionError;
+import static org.assertj.core.util.Lists.list;
+import static org.assertj.core.util.Lists.newArrayList;
+import static org.assertj.core.util.Sets.*;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+
+class IteratorAssert_isUnmodifiable_Test {
+
+  @Test
+  void should_fail_if_actual_is_null() {
+    // GIVEN
+    Iterator<?> actual = null;
+    // WHEN
+    AssertionError assertionError = expectAssertionError(() -> assertThat(actual).isUnmodifiable());
+    // THEN
+    then(assertionError).hasMessage(shouldNotBeNull().create());
+  }
+
+  @ParameterizedTest
+  @MethodSource({"modifiableIterators", "startedModifiableIterators"})
+  void should_fail_if_actual_can_be_modified(/* GIVEN */ Iterator<?> actual, ErrorMessageFactory errorMessageFactory) {
+    // WHEN
+    AssertionError assertionError = expectAssertionError(() -> assertThat(actual).isUnmodifiable());
+    // THEN
+    then(assertionError).as(actual.getClass().getName())
+      .hasMessage(errorMessageFactory.create());
+  }
+
+  // Same as CollectionAssert_isUnmodifiable_Test#modifiableCollections
+  private static Stream<Arguments> modifiableIterators() {
+    return Stream.of(arguments(new ArrayList<>().iterator(), shouldBeUnmodifiable("Iterator.remove()",  new IllegalStateException())),
+      arguments(new LinkedHashSet<>().iterator(), shouldBeUnmodifiable("Iterator.remove()",  new IllegalStateException())),
+      arguments(new LinkedList<>().iterator(), shouldBeUnmodifiable("Iterator.remove()",  new IllegalStateException())),
+      arguments(new HashSet<>().iterator(), shouldBeUnmodifiable("Iterator.remove()",  new IllegalStateException())),
+      arguments(newArrayList(new Object()).iterator(), shouldBeUnmodifiable("Iterator.remove()",  new IllegalStateException())),
+      arguments(newLinkedHashSet(new Object()).iterator(), shouldBeUnmodifiable("Iterator.remove()",  new IllegalStateException())),
+      arguments(newTreeSet("element").iterator(), shouldBeUnmodifiable("Iterator.remove()", new IllegalStateException())));
+  }
+
+  // If Iterator.remove() is called Iterator.next(), it doesn't throw exception
+  private static Stream<Arguments> startedModifiableIterators() {
+    Iterator<?> startedIterator = new ArrayList<>(List.of(1,2,3)).iterator();
```

Comment:
The codebase is still targeting Java 8 so `List.of` cannot be used. You could use `org.assertj.core.util.Lists.list()` (statically imported):

```suggestion
    Iterator<?> startedIterator = new ArrayList<>(list(1,2,3)).iterator();
```